### PR TITLE
composefs: prefer std library functions

### PR DESCRIPTION
**Pull request by giuseppe**
a couple of cleanups for the composefs backend

### CODE REVIEW

**Comment by giuseppe** [comment]
we use "ro" in composefs as well: https://github.com/containers/composefs/blob/main/libcomposefs/lcfs-mount.c#L582-L585, but MS_RDONLY should be enough

**Comment by giuseppe** [comment]
opened a PR for composefs as well: https://github.com/containers/composefs/pull/309

@cgwalters PTAL

File: drivers/overlay/composefs.go [hunk 1]
```diff
@@ -114,23 +110,30 @@ struct lcfs_erofs_header_s {
...
-		return false, &fs.PathError{Op: "openat", Path: path, Err: err}
+		return false, err
 	}
-	defer unix.Close(fd)
+	defer file.Close()
+
 	// do not worry about checking the magic number, if the file is invalid
 	// we will fail to mount it anyway
-	flags := make([]byte, 4)
-	nread, err := unix.Pread(fd, flags, 8)
```
Comment (cgwalters) [inline drivers/overlay/composefs.go:127]: I don't really care too much, but I would note that I think we could use https://pkg.go.dev/os#File.ReadAt here instead

**Review by cgwalters — APPROVED** [review]

**Review by mtrmac — COMMENTED** [review]
LGTM (I didn’t look into the `MS_RDONLY` vs `"ro"` difference, if any).

**Pull request merged**